core/chains/evm/gas/rollups/custom_calldata_da_oracle_test.go

```diff
@@ -0,0 +1,69 @@
+package rollups
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config/oracletype"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/gas/rollups/mocks"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+func TestCustomCalldataDAOracle_NewCustomCalldata(t *testing.T) {
+	oracleAddress := utils.RandomAddress().String()
+	t.Parallel()
+
+	t.Run("panics if oracle type is not CustomCalldata", func(t *testing.T) {
+		ethClient := mocks.NewL1OracleClient(t)
+		daOracleConfig := CreateTestDAOracle(t, oracletype.OPStack, oracleAddress, "")
+		_, err := NewCustomCalldataDAOracle(logger.Test(t), ethClient, daOracleConfig)
+		require.Error(t, err)
+	})
+}
+
+func TestCustomCalldataDAOracle_getCustomCalldataGasPrice(t *testing.T) {
+	oracleAddress := utils.RandomAddress().String()
+	t.Parallel()
+
+	t.Run("correctly fetches gas price if chain has custom calldata", func(t *testing.T) {
+		ethClient := mocks.NewL1OracleClient(t)
+		expectedPriceHex := "0x0000000000000000000000000000000000000000000000000000000000000032"
+
+		daOracleConfig := CreateTestDAOracle(t, oracletype.CustomCalldata, oracleAddress, "0x0000000000000000000000000000000000001234")
+		oracle, err := NewCustomCalldataDAOracle(logger.Test(t), ethClient, daOracleConfig)
+		require.NoError(t, err)
+
+		ethClient.On("CallContract", mock.Anything, mock.IsType(ethereum.CallMsg{}), mock.IsType(&big.Int{})).Run(func(args mock.Arguments) {
+			callMsg := args.Get(1).(ethereum.CallMsg)
+			blockNumber := args.Get(2).(*big.Int)
+			require.NotNil(t, callMsg.To)
+			require.Equal(t, oracleAddress, callMsg.To.String())
+			require.Nil(t, blockNumber)
+		}).Return(hexutil.MustDecode(expectedPriceHex), nil).Once()
+
+		price, err := oracle.getCustomCalldataGasPrice(tests.Context(t))
+		require.NoError(t, err)
+		require.Equal(t, big.NewInt(50), price)
+	})
+
+	t.Run("throws error if custom calldata fails to decode", func(t *testing.T) {
+		ethClient := mocks.NewL1OracleClient(t)
+
+		daOracleConfig := CreateTestDAOracle(t, oracletype.CustomCalldata, oracleAddress, "0xblahblahblah")
+		oracle, err := NewCustomCalldataDAOracle(logger.Test(t), ethClient, daOracleConfig)
+		require.NoError(t, err)
+
+		_, err = oracle.getCustomCalldataGasPrice(tests.Context(t))
+		require.Error(t, err)
+	})
```

Comment:
Just to cover every path could you also add a test for the actual `CallContract` failing too?